docs/graph-algorithms/kosarajus-algorithm.md

```diff
@@ -0,0 +1,299 @@
+
+  For example: 
+  1. Starting from vertex-0, go to vertex-1, vertex-2, and then to vertex-3. Vertex-3 leads to already visited vertex-0, so push the source vertex into the stack.
+
+    [![graph-main.png](https://cdn.programiz.com/sites/tutorial2program/files/scc-step-1.png)](https://postimg.cc/Q9nzFp23)
+
+  2. Go to the previous vertex and visit its child vertices i.e. vertex-4, vertex-5, vertex-6 and vertex-7 sequentially. Since there is nowhere to go from vertex-7, push it into the stack. 
+  
+    [![graph-main.png](https://cdn.programiz.com/sites/tutorial2program/files/scc-step-2.png)](https://postimg.cc/Q9nzFp23)
+
+  3. Go to the previous vertex and visit its child vertices. But, all of its child vertices are visited, so push it into the stack. 
+    [![graph-main.png](https://cdn.programiz.com/sites/tutorial2program/files/scc-step-3.png)](https://postimg.cc/Q9nzFp23)
+
+  4. Similarly, a final stack is created. 
+    [![graph-main.png](https://cdn.programiz.com/sites/tutorial2program/files/scc-step-4.png)](https://postimg.cc/Q9nzFp23)
+
+
+- Reverse the original graph.
+
+    [![graph-1.png](https://cdn.programiz.com/sites/tutorial2program/files/scc-reversed-graph.png)](https://postimg.cc/yk2XHcsL)
+
+- Perform dfs on the reversed graph
+
+  1. Pop vertex-0 from the stack. Starting from vertex-0, traverse through its child vertices (vertex-0, vertex-1, vertex-2, vertex-3 in sequence) and mark them as visited. The child of vertex-3 is already visited, so these visited vertices form one strongly connected component. 
+    [![graph-main.png](https://cdn.programiz.com/sites/tutorial2program/files/scc-reversed-step-1.png)](https://postimg.cc/Q9nzFp23)
+
+  2. Go to the stack and pop the top vertex if already visited. Otherwise, choose the top vertex from the stack and traverse through its child vertices as presented above. 
+    [![graph-main.png](https://cdn.programiz.com/sites/tutorial2program/files/scc-final-graph.png)](https://postimg.cc/Q9nzFp23)
+
+
+- Now, the strongly connected components are:
+
+  [![graph-main.png](https://cdn.programiz.com/sites/tutorial2program/files/scc-step-3.png)](https://postimg.cc/Q9nzFp23)
+
+
+## Algorithm
+
+1. For each vertex u, mark u as unvisited. Let L be empty.
+
+2. For each vertex u, do Visit(u), where Visit(u) is the recursive subroutine:
+    - If u is unvisited then:
+      1. Mark u as visited.
+      2. For each out-neighbour v of u, do Visit(v).
+      3. Prepend u to L.
+  Otherwise do nothing.
+
+3. For each element u of L in order, do Assign(u,u) where Assign(u,root) is the recursive subroutine:
+    - If u has not been assigned to a component then:
+      1. Assign u as belonging to the component whose root is root.
+      2. For each in-neighbour v of u, do Assign(v,root).
+
+
+## Code
+
+### cpp
```

Comment:
That's not how you should add Code. Check the existing articles to know how to do it

Reply:
Updated.